File: Cpp/978_Longest_Turbulent_Subarray.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& A) {
        if (A.size() < 2) return A.size();
        int result = 0;
        int count = 0;
        int direction = 0;
        for(int i=1; i<A.size(); ++i){
            if(A[i-1] - A[i] == 0){
                count = 0;
                direction = 0;
            } else if(A[i-1] - A[i] > 0){
                count = direction < 0 ? count + 1 : 1;
                direction = 1;
            } else{
                count = direction > 0 ? count + 1 : 1;
                direction = -1;
            }
            result = max(result, count);
        }
        
        return result + 1;
    }
};

File: Kotlin/978_Longest_Turbulent_Subarray.kt
class Solution {
    fun maxTurbulenceSize(arr: IntArray): Int {
        if (arr.size < 2) return arr.size
        
        var result = 0
        var count = 0
        var direction = 0
        
        for(i in 1 until arr.size){
            if(arr[i-1] - arr[i] == 0){
                count = 0;
                direction = 0
            } else if(arr[i-1] - arr[i] > 0){
                count = if(direction < 0) count + 1 else 1
                direction = 1
            } else{
                count = if(direction > 0) count + 1 else 1
                direction = -1
            }
            result = maxOf(result, count)
        }
        
        return result + 1 
    }
}